Tools/CISupport/ews-build/results_db.py

```diff
@@ -88,13 +88,78 @@ def get_results_summary(cls, test, commit=None, configuration=None, logger=None,
 
         defer.returnValue({})
 
+    @classmethod
+    @defer.inlineCallbacks
+    def get_suite_results(cls, suite, commit=None, configuration=None, logger=None):
+        logger = logger or (lambda log: None)
+        params = dict()
+        suite = suite or cls.DEFAULT_SUITE
+        if suite not in cls.SUITES:
+            logger(f"'{suite}' is not a valid suite name\n")
+            defer.returnValue({})
+            return
+        if not configuration:
+            configuration = {}
+        for key, value in configuration.items():
+            if key not in cls.CONFIGURATION_KEYS:
+                logger(f"'{key}' is not a valid configuration key\n")
+            params[key] = value
+        if commit:
+            params['ref'] = commit
+
+        response = yield TwistedAdditions.request(f'{cls.HOSTNAME}/api/results/{urllib.parse.quote(suite)}', params=params, logger=logger)
+        if not response:
+            logger(f'No response from {cls.HOSTNAME}\n')
+        elif response.status_code == 200:
+            defer.returnValue(response.json())
+            return
+        else:
+            logger(f'Failed to query results summary with status code {response.status_code}\n')
+
+        defer.returnValue({})
+
+    @classmethod
+    @defer.inlineCallbacks
+    def get_results(cls, test, commit=None, configuration=None, logger=None, suite=None, default=None):
```

Comment:
I think `get_results`, `get_suite_results ` and `get_results_summary` can share much more code. They're all basically identical save the URL endpoint they're hitting, right?

Reply:
Good point! I combined `get_results` and `get_suite_results` into one function and made `test` an optional argument. Now `get_results` and `get_results_summary` call `make_request` by passing in their arguments plus which endpoint to use.